mautic/mautic
# Remove deprecations from ChannelBundle

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? |
| New feature? | N
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/7992
| BC breaks? | Y
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR removes deprecated code from ChannelBundle.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. There is no bug so there is nothing to reproduce. 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Open `Marketing Messages`.
3. Try to  to create and send some messages. You should see no errors.

#### List backwards compatibility breaks:
1. Removed Mautic\ChannelBundle\ModelMessageQueueModel::dispatchDeprecatedEvent method
The events it dispatches have no listeners.
2. Removed Mautic\ChannelBundle\ChannelEvents::ON_CAMPAIGN_TRIGGER_ACTION event.
It has no listeners and is deprecated.
3. Removed Mautic\LeadBundle\Event\ChannelEvent class.
It's not in the ChannelBundle but it extends one of its classes and is deprecated. It easier to remove it in this PR.
4. Removed Mautic\ChannelBundle\Event\ChannelEvent::setChannel and Mautic\ChannelBundle\Event\ChannelEvent::getChannels
